src/main/java/nextstep/courses/domain/session/Session.java

```diff
@@ -0,0 +1,43 @@
+package nextstep.courses.domain.session;
+
+import nextstep.courses.CannotEnrollException;
+import nextstep.courses.strategy.PaymentStrategy;
+import nextstep.payments.domain.Payment;
+import nextstep.users.domain.NsUser;
+
+import java.time.LocalDate;
+
+public class Session {
+    private final SessionImage image;
+    private final SessionDate date;
+    private final SessionState state;
+    private final Enrollment enrollment;
+    private final PaymentStrategy paymentStrategy;
+
+    public Session(SessionImage image, SessionDate date, SessionState state, Enrollment enrollment, PaymentStrategy paymentStrategy) {
+        this.image = image;
+        this.date = date;
+        this.state = state;
+        this.enrollment = enrollment;
+        this.paymentStrategy = paymentStrategy;
+    }
+
+    public void applySession(NsUser user, LocalDate enrollDate, Payment payment) {
+        if (!canApply(enrollDate, payment)) {
+            throw new CannotEnrollException("등록 불가능한 상태입니다.");
+        }
+        this.enrollment.enroll(user);
+    }
+
+    private boolean canApply(LocalDate enrollDate, Payment payment) {
+        if (!state.canRecruit()) {
```

Comment:
```suggestion
        if (state.cannotRecruit()) {
```

`!`표현 보다는 있는 그대로 메소드명으로 표현하는게 가독성에 좋을수도 있다고생각되네요 🤔